docs/api-guide.md

```diff
@@ -894,60 +960,61 @@ public class Sample {
+- If you catch `CommitConflictException`, this exception indicates that committing the transaction has failed due to transient faults (for example, a conflict error). In this case, you can retry the transaction from the beginning.
+- If you catch `UnknownTransactionStatusException`, this exception indicates that the status of the transaction, whether it was successful or not, is unknown. In this case, you need to check if the transaction is committed successfully and retry the transaction if it has failed.
 
-Please note that if you begin a transaction by specifying a transaction ID, you must use a different ID when you retry the transaction.
-And, in the sample code, the transaction is retried three times maximum and sleeps for 100 milliseconds before it is retried.
-But you can choose a retry policy, such as exponential backoff, according to your application requirements.
+How to identify a transaction status is delegated to users. You may want to create a transaction status table and update it transactionally with other application data so that you can get the status of a transaction from the status table.
 
-## Transactional operations for Two-phase Commit Transaction
+### Notes about some exceptions
 
-Please see [Two-phase Commit Transactions](two-phase-commit-transactions.md).
+Although not illustrated in the example code, the `resume()` API could also throw `TransactionNotFoundException`. This exception indicates that the transaction associated with the specified ID was not found and/or the transaction might have expired. In either case, you can retry the transaction from the beginning since the cause of this exception is basically transient.
 
-## Investigate Consensus Commit transactions errors
+In the sample code, for `UnknownTransactionStatusException`, the transaction is not retried because the cause of the exception is non-transient. For other exceptions, the transaction is retried because the cause of the exception is transient or non-transient. If the cause of the exception is transient, the transaction may succeed if you retry it. However, if the cause of the exception is non-transient, the transaction will still fail even if you retry it. In such a case, you will exhaust the number of retries.
 
-This configuration is only available to troubleshoot Consensus Commit transactions. By adding the following configuration, `Get` and `Scan` operations results will contain [transaction metadata](schema-loader.md#internal-metadata-for-consensus-commit).
-To see the transaction metadata columns details for a given table, you can use the `DistributedTransactionAdmin.getTableMetadata()` method which will return the table metadata augmented with the transaction metadata columns. 
-All in all, using this configuration can be useful to investigate transaction related issues.
+{% capture notice--info %}
+**Note**
+
+If you begin a transaction by specifying a transaction ID, you must use a different ID when you retry the transaction.
```

Comment:
I think we should probably say this in the `begin` section (as I commented) instead of here.

Reply:
Got it. I've removed that sentence from this note since I've added it to the `Begin or start a transaction` section.